src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx

```diff
@@ -161,11 +161,8 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
     }
 
     if (shapeToolData?.supertypes?.behavior?.item?.schematicDesignation) {
-      if (Array.isArray(shapeToolData?.supertypes?.behavior?.item?.schematicDesignation)) {
-        setSchematicDesignation(shapeToolData?.supertypes?.behavior?.item?.schematicDesignation.join(", "));
-      } else {
-        setSchematicDesignation(shapeToolData?.supertypes?.behavior?.item?.schematicDesignation);
-      }
+      const schematicDesignations = asArray(shapeToolData?.supertypes?.behavior?.item?.schematicDesignation);
```

Comment:
@Kasmadei i thought you said asArray is not possible to use here so i suggested to change the method by copying it from other project, see https://github.com/kbss-cvut/fta-fmea-ui/issues/361#issuecomment-2153651337